### ISSUE TITLE
Can't generate unique value in object.

### ISSUE THREAD

**Issue by mandaputtra** [json-schema-faker/json-schema-faker#598]
Is there any feature that can make some properties in object unique? I have look this [issue](https://github.com/json-schema-faker/json-schema-faker/issues/291), but seems the solution are using increment therefore in my case I'm using predefined string. Any idea what should I use?

1. the JSON Schema content that fails - preferably using http://json-schema-faker.js.org/ (paste the schema, click `Generate example` and then copy and paste the resulting URL from the browser address bar)

```json
{
  "type": "array",
  "minItems": 2,
  "maxItems": 3,
  "items": {
    "name": {
      "type": "string",
      "unique": true,
      "oneOf": [{ "const": "pairs", "description": "hh" }, { "const": "jj", "description": "hw" }, { "const": "pp", "description": "2h" }, { "const": "kk", "description": "hf" }]
    },
    "email": {
      "type": "string",
      "faker": "internet.email"
    }
  },
  "required": [
    "name",
    "email"
  ]
}
```

2. the result you got

```json
[
  {
    "name": "pp",
    "email": "Kira_Lynch81@gmail.com"
  },
  {
    "name": "pp",
    "email": "Adrianna85@hotmail.com"
  },
  {
    "name": "pp",
    "email": "Vance29@yahoo.com"
  }
]
```

3. the result you expected

All the `name` value is differen, and based on `oneOf` value.

**Comment by pateketrueke** [comment]
Yeah, the `unique` keyword on object properties is not supported yet, I think we could implement that using `jsf.define()` but currently oneOf/allOf/anyOf references are pre-resolved before custom keywords.

I think we should run custom keywords before other references are resolved, otherwise we wouldn't be able to solve this.

So much thanks for your input!
